src/DonationForms/resources/app/hooks/useDonationSummary.ts

```diff
@@ -1,22 +1,118 @@
 import {
-    DonationSummaryLineItem,
+    DonationSummaryLineItem, DonationTotals,
     useDonationSummaryContext,
-    useDonationSummaryDispatch,
+    useDonationSummaryDispatch
 } from '@givewp/forms/app/store/donation-summary';
 import {
     addAmountToTotal,
     addItem,
     removeAmountFromTotal,
-    removeItem,
+    removeItem
 } from '@givewp/forms/app/store/donation-summary/reducer';
 import {useCallback} from '@wordpress/element';
+import type {
+    subscriptionPeriod
+} from '@givewp/forms/registrars/templates/groups/DonationAmount/subscriptionPeriod';
 
 /**
+ * Zero decimal currencies are currencies that do not have a minor unit.
+ * For example, the Japanese Yen (JPY) does not have a minor unit.
+ * @unreleased
+ *
+ * @see https://stripe.com/docs/currencies#zero-decimal
+ */
+const zeroDecimalCurrencies = [
+    'BIF',
+    'CLP',
+    'DJF',
+    'GNF',
+    'JPY',
+    'KMF',
+    'KRW',
+    'MGA',
+    'PYG',
+    'RWF',
+    'UGX',
+    'VND',
+    'VUV',
+    'XAF',
+    'XOF',
+    'XPF',
+];
```

Comment:
@jonwaldstein I recommend using the following array to also identify 3-decimal currencies:
```php
$nonTwoDecimalCurrencies = [
    'BIF' => 0,
    'CLP' => 0,
    'DJF' => 0,
    'GNF' => 0,
    'ISK' => 0,
    'JPY' => 0,
    'KMF' => 0,
    'KRW' => 0,
    'PYG' => 0,
    'RWF' => 0,
    'UGX' => 0,
    'VND' => 0,
    'VUV' => 0,
    'XAF' => 0,
    'XOF' => 0,
    'XPF' => 0,
    'BHD' => 3,
    'IQD' => 3,
    'JOD' => 3,
    'KWD' => 3,
    'LYD' => 3,
    'OMR' => 3,
    'TND' => 3,
];
```

Reply:
You could then adjust your math:
```js
decimals = $nonTwoDecimalCurrencies[$currency] ?? 2;
return Math.round(parseFloat(amount) * (10 ** decimals)); 
```